File: Server/AllClass/RoleState.js
var skillList = require("../Config/SkillList");

class RoleState {
    
    constructor() {

        // 战前相关
        this.sid = "";
        this.username = "";
        this.roleName = "";
        this.level = 1;
        this.exp = 0;
        this.rank = 0;
        this.score = 0;
        this.gold = 0;
        this.inServerId = -1;
        this.waitMode = -1;

        // 战斗时相关（不会存入数据库）
        this.inFieldId = -1;
        this.lifeOrigin = 5.0;
        this.life = 5.0;
        this.blockLifeOrigin = 3.0;
        this.blockLife = 3.0;
        this.isBlocking = false;
        this.towerLifeOrigin = 50.0;
        this.towerLife = 50.0;

        // 技能
        this.currentSpell = "";
        this.skillList = skillList;
        // this.blockSkill = new BuffSkill(1);
        // this.normalBullet = new BulletSkill(2);
        // this.slowBullet = new BulletSkill(3);
        // this.fastBullet = new BulletSkill(4);
        // this.rayBullet = new BulletSkill(5);
        // this.blockWall = new BuffSkill(6);
        // this.shadow = new BuffSkill(7);
        // this.shield = new BuffSkill(8);

        // 状态
        this.isLeftAlly = true;
        this.isSinging = false;
        this.isImmune = false;
        this.isReflect = false;
        this.blockCount = 0;
        this.isDead = false;

    }
}

module.exports = RoleState;

File: Gun_Block/Assets/Script/Factory/State/RoleState.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[Serializable]
public class RoleState{

    // ———— 战前相关 ————
    public string sid;
    public string username;
    public string roleName;
    public int level;
    public int exp;
    public int rank;
    public int score;
    public int gold;
    public int inServerId = -1;
    public int waitMode = -1;
    
    // ———— 战时相关 ————（不存入数据库）
    public string oppoSid = ""; // 对位sid
    public int inFieldId = -1;
    public float lifeOrigin = 5;
    public float life = 5;
    public float blockLifeOrigin = 3;
    public float blockLife = 3;
    public bool isBlocking = false;
    
    // ———— 技能 ————
    public string currentSpell = ""; // 技能命令
    public SkillState[] skillList = null; // 技能表
    // public BuffSkill blockSkill; // 格挡 AAA
    // public BulletSkill normalBullet; // 普攻
    // public BulletSkill slowBullet; // 慢攻
    // public BulletSkill fastBullet; // 快攻
    // public BulletSkill rayBullet; // 激光
    // public BuffSkill blockWall; // 拒鹿
    // public BuffSkill shadow; // 雾
    // public BuffSkill shield; // 粒子盾

    // ———— 状态 ————
    public bool isLeftAlly = true; // 阵营
    public bool isSinging = false;
    public bool isImmune = false;
    public bool isReflect = false;
    public int blockCount = 0;
    public bool isDead = false; // 复活状态

}